pkgs/sketch_pad/lib/flutter_samples.dart

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:http/http.dart' as http;
+
+const String _stableApiDocsUrl = 'https://api.flutter.dev/snippets';
+const String _mainApiDocsUrl = 'https://main-api.flutter.dev/snippets';
+
+class FlutterSampleLoader {
+  final http.Client client = http.Client();
+
+  Future<String> loadFlutterSample({
+    required String sampleId,
+    String? channel,
+  }) async {
+    // There are only two hosted versions of the docs: master and stable.
+    final sampleUrl = channel == 'master'
```

Comment:
I'd check for either "master" or "main" here, since eventually we will probably rename it. In fact, checking for both here will *allow* it to be renamed.